packages/frontend/src/types.ts

```diff
@@ -57,11 +56,10 @@ export interface AugmentedPriceRangeSpecification {
 }
 
 export interface CampaignPayload {
+    targetType?: TargetType;
     rewardType?: RewardType;
```

Comment:
It's tough for me to understand why we need these top level things when this information is kinda encoded already in the `distributables` and `target` keys.

Reply:
The issue here is that the `target` encapsulate both the dex and the pool info (all required), and thus it can't be initialized in the form at the start, once the campaign type gets chosen; also in the form `dex` and `pool` (that together are the `target`) are separated and the `dex` comes first, so the `target` can be undefined while the dex has been selected; this wasn't quite working with the type just like the sdk, and also having partial types was king of a mess, so I opted for these top level props. Do you think there's a better way?

Reply:
With this in mind, I definitely think this can work